src/compiler/checker.ts

```diff
@@ -23947,18 +23948,26 @@ namespace ts {
             /**
              * Copy the given symbol into symbol tables if the symbol has the given meaning
              * and it doesn't already existed in the symbol table
-             * @param key a key for storing in symbol table; if undefined, use symbol.name
              * @param symbol the symbol to be added into symbol table
              * @param meaning meaning of symbol to filter by before adding to symbol table
              */
             function copySymbol(symbol: Symbol, meaning: SymbolFlags): void {
                 if (getCombinedLocalAndExportSymbolFlags(symbol) & meaning) {
                     const id = symbol.escapedName;
-                    // We will copy all symbol regardless of its reserved name because
-                    // symbolsToArray will check whether the key is a reserved name and
-                    // it will not copy symbol with reserved name to the array
+                    if (isReservedMemberName(id)) {
+                        return;
+                    }
+
                     if (!symbols.has(id)) {
-                        symbols.set(id, symbol);
+                        symbols.set(id, symbol.flags);
+                        results.push(symbol);
+                    }
+                    else {
+                        const targetFlag = symbols.get(id);

```

Comment:
Instead of having symbols.has and symbols.get just make one call using:
```typescript
const flags = symbols.get(id);
if (flags === undefined) {
}
else {
}
```